.github/workflows/test-sample-apps.yml

```diff
@@ -64,15 +66,27 @@ jobs:
         run: |
           echo "### Testing HTTP2_APP app ###"
           yq e '.applications.[].name' "${GITHUB_WORKSPACE}/http2/apps-manifest.yml" | grep -i 'http2' | while read -r HTTP2_APP; do
-              echo "curl -v --http2-prior-knowledge -H 'Connection: close' 'https://$HTTP2_APP.${{ secrets.CF_DOMAIN }}'"
-              curl -v --http2-prior-knowledge -H "Connection: close" "https://$HTTP2_APP.${{ secrets.CF_DOMAIN }}"
+              ROUTE=${{ github.run_id }}-$HTTP2_APP
+              cf8 map-route $HTTP2_APP ${{ secrets.CF_DOMAIN }} --hostname $ROUTE --app-protocol http2
+              echo "curl -v --http2-prior-knowledge -H 'Connection: close' 'https://$ROUTE.${{ secrets.CF_DOMAIN }}'"
+              curl -v --http2-prior-knowledge -H "Connection: close" "https://$ROUTE.${{ secrets.CF_DOMAIN }}"
           done
           echo "Testing GRPC apps"
-          grpcurl go-grpc-test.${{ secrets.CF_DOMAIN }}:443 example.Example.Run
-          grpcurl -proto http2/java-grpc/app/src/main/proto/example.proto java-grpc-test.${{ secrets.CF_DOMAIN }}:443 example.Example.Run 
-          grpcurl -proto http2/node-grpc/example.proto node-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run 
-          grpcurl -proto http2/python-grpc/example.proto python-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run
-          grpcurl -proto http2/ruby-grpc/example.proto ruby-grpc-test.${{ secrets.CF_DOMAIN }}:443 Example.Run
+          ROUTE=${{ github.run_id }}-go-grpc-test
+          cf8 map-route go-grpc-test ${{ secrets.CF_DOMAIN }} --hostname $ROUTE --app-protocol http2
+          grpcurl $ROUTE.${{ secrets.CF_DOMAIN }}:443 example.Example.Run
+          ROUTE=${{ github.run_id }}-java-grpc-test
+          cf8 map-route java-grpc-test ${{ secrets.CF_DOMAIN }} --hostname $ROUTE --app-protocol http2
+          grpcurl -proto http2/java-grpc/app/src/main/proto/example.proto $ROUTE.${{ secrets.CF_DOMAIN }}:443 example.Example.Run 
```

Comment:
the only thing different is the path to the proto files. These files are all identical and just kept with the examples for completeness. There is no good reason to explicitly use different ones. In fact it would be better to use a single one to ensure that the examples all do the same thing and not something different.

Reply:
In fact, the `grpcurl` command for the `go-grpc-test` app differs from the remaining ones. I'd like to keep the focus on fixing the route issue in this Pull-Requst, and do other improvements separately.